content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+
+### Introduction
+
+In any modern system, data handling is one of the major and vital goals. The way data is handled is what makes the total difference. Some systems go for generic one which includes lists, dictionaries, and hash.
+
+Others prefer non-generic which are arrays, array lists, and bit arrays. Non- generic programming represents an ordered collection of an object that can be indexed individually. As a matter of fact, generic programming is strongly typed while non- generic programming is not strongly typed, which means generic programming enforces strict rules on blending of values with different data types.
+
+But due to the great advancement of technology in the 21st century, many software systems require to store large groups of entities
+rather than individual entry. The concept of non-generic data handling can't fully solve this problem therefore in C#, generic collections are designed to solve this problem. Generic is a class that gives users the ability to define classes and methods with the placeholder.
+
+### Collections
+**Collections** according to C# are classes that hold together groups of objects. The `.NET` libraries provide useful system-defined classes, like collection namespace, which provide various collections and their functionalities. The use of non-generic collections is normally discouraged therefore my article fully focuses on generic collections.
+
+We have three common generic collections incorporated in the C# namespace.
+
+## 1. LIST
+
+A list is the most commonly used generic collection. Its because they are much more convenient when they are used instead of an array.
+A list is a very general data structure with each entry occupying a specific position known as an index. Lists are very flexible as they automatically resize every time data is added.
+
+Lists are very easy to manipulate. This is because many methods have been created and abstracted, meaning the bulky work is already done for the user. The user only has to call the method and manipulate the list.
+ 
+A list stores data in sequence, although the sequence oughtn't to have to be particular. It can also accept duplicate entries.
+Below is a pictorial explanation of a list. A generic list class is declared as shown:
+
+```
+public class genericList<T>
+
+{
+    
+    void add(T input){}
+}
+```
+
+On the same note, a list can be of integer or string data type. They are declared differently but the syntax is the same as shown:
+```
+using system;
+
+{
+private class genericListExample
+static void Main ()
+{
+    genericlist<int> list1 = new genericlist<int>(); //This declares a  list of type int.
```

Comment:
> //This declares a  list of type int.

Remove the second space after *a* and before *list*. Also, I personally would add a space between the *//* and *This* but as long as you are consistent it should be fine.